Sources/BraintreeCore/ConfigurationCache.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+class ConfigurationCache {
+        
+    // MARK: - Private Properties
+    
+    private let timeToLiveMinutes = 5.0
+    
+    // MARK: - Internal Properties
+    
+    /// Exposed for testing
+    let cacheInstance = NSCache<NSString, BTConfiguration>()
+
+    // MARK: - Singleton Setup
+    
+    static let shared = ConfigurationCache()
+    private init() { }
```

Comment:
I think since we are providing defaults to the properties this can be removed, right?

Reply:
The `private init()`? This prevents other parts of the codebase from being able to instantiate an instance of `ConfigurationCache`, making it a true singleton.